packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -99,11 +99,15 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
 
       cy.state('latestActiveDomain', domain)
 
-      return new Bluebird((resolve, reject) => {
+      return new Bluebird((resolve, reject, onCancel) => {
         const cleanup = () => {
           communicator.off('queue:finished', onQueueFinished)
```

Comment:
Doesn't need to be done on this PR but are we also stacking up other events besides `queue:finished`? Seems like we should be unregistering the other events as well.

Reply:
I wonder if there are any scenarios that cause `sync:globals` to get stacked?

Reply:
Yes, any of these listeners have the potential to leak if not called once. I will investigate further, but in another PR.